compiler/ccgstmts.nim

```diff
@@ -1147,5 +1147,7 @@ proc genAsgn(p: BProc, e: PNode, fastAsgn: bool) =
 
 proc genStmts(p: BProc, t: PNode) =
   var a: TLoc
+  pushInfoContext(p.config, t.info)
```

Comment:
The backend should not be concerned with the quality of error messages.

Reply:
without this, I get one of the 2 bugs I mentioned